mantidproject/mantid
# New algorithm to load elemental analysis data

**Description of work.**

This PR is for a new algorithm - LoadElementalAnalysisData - and associated system and unit tests and documentation. The new algorithm will read elemental analysis data and add the appropriate Group Workspace and Workspaces that correspond to the run number provided.

The algorithm does not overwrite load.

Added funtionality compared to load_utils.py
- User receives a message if files for the run number provided cannot be located
- Plots have a key

The algorithm is the first stage in revising the Elemental Analysis GUI.

**To test:**

1. Code Review
2. Download & unzip the 12 dat files for run 9999 (held in attached .zip folder)
[ral09999.rooth5099.zip](https://github.com/mantidproject/mantid/files/5402848/ral09999.rooth5099.zip)
3. Run LoadElementalAnalysisData algorithm with Run 9999 and GroupWorkspace 9999
4. A GroupWorkspace called 9999 containing 4 Workspaces ['9999; Detector 1', '9999; Detector 2', '9999; Detector 3',  '9999; Detector 4'] should become available
5. Right click Workspace '9999; Detector 1' and click Plot->Spectrum
6. Select Plot All
7. The following plot should appear
![Screenshot (49)](https://user-images.githubusercontent.com/55837273/96471146-e978d700-1226-11eb-9bc4-e334c0282b52.png)


Fixes #29519 


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
